opendevin/events/stream.py

```diff
@@ -21,9 +25,32 @@ class EventSource(str, Enum):
 
 
 class EventStream:
-    _subscribers: dict[str, Callable] = {}
-    _events: list[Event] = []
+    sid: str
+    _subscribers: dict[str, Callable]
+    _events: list[Event]
     _lock = asyncio.Lock()
+    _file_store: FileStore
+
+    def __init__(self, sid: str):
+        self.sid = sid
+        self._file_store = get_file_store()
+        self._subscribers = {}
+        self._events = []
+
+    def _get_filename_for_event(self, event: Event):
+        # TODO: change to .id once that prop is in
+        return f'sessions/{self.sid}/events/{event._id}.json'  # type: ignore [attr-defined]
+
+    async def _rehydrate(self):
+        async with self._lock:
+            self._events = []
+            events = self._file_store.list(f'sessions/{self.sid}/events')
+            print('list of events', events)
+            for event_str in events:
+                content = self._file_store.read(event_str)
+                data = json.loads(content)
+                event = event_from_dict(data)
+                self._events.append(event)
 
     def subscribe(self, id: EventStreamSubscriber, callback: Callable):
```

Comment:
Hi Robert, the current subscribe/unsubscribe strategy is based on `agent_controller`, `server`, `runtime`, `main` enum. It's not based on the instance. 
So  one OpenDevin server can't serve two sessions(two users or two browsers chat with opendevin) at the same time right? The second subscribe call will be rejected ('Subscriber subscribed multiple times: ' + id) .